[CS2113-T14-2] eCardnomics

kaijiel24 opened the pull request:
Flash Card manager for Economics students on Command Line

alwaysnacy committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

alwaysnacy committed:
add Tag-DG

alwaysnacy committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into trang-UG-DG

LiewWS committed:
Remove unused method and refactor delete branch for SLAP

LiewWS committed:
Update .gitignore to not track data/deckList.txt

alwaysnacy committed:
add TagFlow

alwaysnacy committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into trang-UG-DG

LiewWS committed:
Add one-shot delete deck command

alwaysnacy committed:
update UG and DG

alwaysnacy committed:
Merge pull request #115 from alwaysnacy/master

update UG and DG

LiewWS committed:
Add one-shot add flashcard command

LiewWS committed:
Move confirmation methods to Ui

Aliciaho commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,258 @@
+
+#### Commands
+
+![DG-Design Commands UML](./images-dg/DG-Design-Commands.png?raw=true "Commands UML Class Diagram")
```
Maybe you would like to break this diagram into two parts, so that it is less complicated?

zhixiangteoh replied:
Changed, thanks!

Aliciaho commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,258 @@
+   interacts with `GameStorage` on package-private basis; i.e., `GameEngine` and `GameStorage` have full mutual
+    access as if they were a single class. This is one of the main intentional design decisions.
+  
+![DG-Implementation-Features-Game-Mode-Architecture](./images-dg/Game-Mode-Design.png?raw=true "Game Mode
```
Maybe you would like to split up this digram into two parts, so that it is less complicated?

zhixiangteoh replied:
Changed, thanks!

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into branch-UserGuide

Merge branch-UserGuide into master

Aliciaho commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,258 @@
+
+The following elaborates the execution flow of Game Mode, from after a `start` command has been parsed in Normal Mode:
+
+![DG-Implementation-Features-Game-Mode-Sequence](./images-dg/Game-Mode-Sequence.png?raw=true "Game Mode UML Sequence
```
Maybe it would be better to simplify this diagram/ split them up into smaller parts, so that it would be easier to follow?

zhixiangteoh replied:
Changed with reference frames, thanks!

LiewWS committed:
add one-shot delete delete command

LiewWS committed:
refactor update for SLAP

teohdd committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

zhixiangteoh committed:
Merge pull request #116 from LiewWS/master

Implement one-liner commands

teohdd committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

LiewWS committed:
update UG to include one-line commands

LiewWS committed:
Merge pull request #117 from LiewWS/branch-UserGuide

Merge branch-UserGuide to master

teohdd committed:
update help command list for all modes

teohdd committed:
fix checkstyle

wangwaynesg committed:
Add DG for Storage class

wangwaynesg committed:
Merge pull request #121 from wangwaynesg/branch-StorageDG

Add DG for Storage class

zhixiangteoh committed:
Merge pull request #119 from zhixiangteoh/master

Update Help commands

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
```
You can consider resizing your diagram such that the text size in the diagram matches the the text size of the main text of the diagram.

![image](https://user-images.githubusercontent.com/60382274/97514153-0c468080-19c9-11eb-9c58-6b195b579ded.png)

e0425705 replied:
Should this be a dotted arrow?

![image](https://user-images.githubusercontent.com/60382274/97515391-e9699b80-19cb-11eb-81b1-2dc1e4517c5b.png)

siewwencode replied:
Should this be a dotted arrow?

![image](https://user-images.githubusercontent.com/60382274/97515391-e9699b80-19cb-11eb-81b1-2dc1e4517c5b.png)

TomLBZ replied:
Is this part necessary? It is neither a method call nor a return value.
![image](https://user-images.githubusercontent.com/26886705/97535965-66f6d100-19f7-11eb-8acd-cafa994efbb1.png)

JunxianAng replied:
Hi maybe a short explanation can be given to help readers understand your sequence diagram better?

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+    * `Deck List` : A complete list of all the `Deck`s in memory.
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
```
Is this notation for the sequence diagram correct?

![image](https://user-images.githubusercontent.com/60382274/97514259-4021a600-19c9-11eb-9ec8-cd07e1e94a90.png)

samuellleow commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+An additional feature targeted at students who wish to use add more style to their flash cards outside of the command 
+line option to allow keep things interesting when they are revising.
+
+The `PowerPointCommand` is parsed by `NormalParser` but the "Print to PowerPoint" command can be called from both Normal 
```
Should there be a return to PowerpointCommand after Powerpoint is created in the sequence diagram?

I believe that the box to contain the "opt" is wrong, you may refer to the notes again, it should be a sharp box with a chip of the corner.

kaijiel24 replied:
@kaijiel24

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+#### Overall Logic
+![DG-Overall Logic UML](./images-dg/Logic-DG.png?raw=true "Overall Logic Diagram")
+
+1. The overall logic component consists of the Parser class and Command class.
+2. The Parser parses the user input and creates the respective Command object.
+3. This command will be executed by the Main class.
```
You might have missed out bolding the classes here

siewwencode replied:
You might have missed out bolding the classes here

flsqqq replied:
As the names of the classes are in-line code, should there be back-ticks around them?
e.g. `Parser` class

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+and Deck Mode. 
+
+The following diagram shows how the `PowerPointCommand`'s `execute()` calls the `createNewPowerPoint()` method of `PowerPoint`.
+![PPTX Sequence Diagram](images-dg/DG-PPTX-Sequence-Diagram.png)
```
Are you missing the return arrows?

![image](https://user-images.githubusercontent.com/60382274/97514645-3ba9bd00-19ca-11eb-8e03-ee2c0a704e6c.png)

siewwencode replied:
Are you missing the return arrows?

![image](https://user-images.githubusercontent.com/60382274/97514645-3ba9bd00-19ca-11eb-8e03-ee2c0a704e6c.png)

TomLBZ replied:
Maybe you want to have return arrows for these method calls just to be consistent with other diagrams on the DG?
![image](https://user-images.githubusercontent.com/26886705/97536361-fac89d00-19f7-11eb-87cb-dcc5f95ba8d4.png)

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+The user can also modify to tags of the decks by using tag or untag command, and uses search by tag to find
+a group of decks he/she is interested in.
+
+![DG-Implementation-Features-TagSequence](./images-dg/Tag.png?raw=true)
```
Are you missing a return arrow here?

![image](https://user-images.githubusercontent.com/60382274/97514907-d4d8d380-19ca-11eb-8d65-cd726d3459e0.png)

samuellleow replied:
![image](https://user-images.githubusercontent.com/60358173/97516053-4285ff00-19cd-11eb-89d6-9f7d90e49316.png)
Should there be a return arrow? Prof said that it is optional if a void function is called

samuellleow commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
+The **Sequence Diagram** above shows how the components interact for a basic `create <deck name>` command where a new deck is created and added in to the `Deck List`.
```
![image](https://user-images.githubusercontent.com/60358173/97515625-66951080-19cc-11eb-8079-ff20a306d8ce.png)
I believe the return arrow should be a dotted line

The arrows coming into the UI class shouldnt be into or from the middle of the box, should be from the top or bottom of the box

kaijiel24 replied:
@kaijiel24

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+   microeconomics`:
+  
+![DG-Design CreateCommand Sequence UML](./images-dg/DG-Design-Sequence-Diagram.png?raw=true "CreateCommand UML
+ Sequence Diagram")
```
You can consider terminating CreateCommand

![image](https://user-images.githubusercontent.com/60382274/97516099-5c274680-19cd-11eb-93e2-c6a6844b1052.png)

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+`toString(boolean isQuestion, int offset)` method of a `FlashCard`
+object.
+
+![DG-Implementation-Features-PP-Sequence](./images-dg/PP-Sequence.png?raw=true)
```
Are you labelling correct? Perhaps it should be toString(isQuestion, offset) and formatResponse(isQuestion, offset)

![image](https://user-images.githubusercontent.com/60382274/97516241-aa3c4a00-19cd-11eb-8e3e-5718f4eec84a.png)

e0425705 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+The following elaborates the execution flow of Game Mode, from after a `start` command has been parsed in Normal Mode:
+
+![DG-Implementation-Features-Game-Mode-Sequence](./images-dg/Game-Mode-Sequence.png?raw=true "Game Mode UML Sequence
```
some return arrows are missing

![image](https://user-images.githubusercontent.com/60382274/97516571-58e08a80-19ce-11eb-90b4-0d4385ccc1c1.png)

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+    * `Deck List` : A complete list of all the `Deck`s in memory.
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
```
This part is a bit confusing? Perhaps you can elaborate more?

![image](https://user-images.githubusercontent.com/60382274/97515807-ca1f3e00-19cc-11eb-9de7-392fd77bcf0c.png)

kaijiel24 replied:
@kaijiel24

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
```
You can consider resizing your diagram such that the text size in the diagram matches the the text size of the main text of the diagram.

![image](https://user-images.githubusercontent.com/60382274/97514153-0c468080-19c9-11eb-9c58-6b195b579ded.png)

kaijiel24 replied:
@kaijiel24

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+  sequence diagram showcasing this interaction, for execution of a `CreateCommand`, e.g. `create
+   microeconomics`:
+  
+![DG-Design CreateCommand Sequence UML](./images-dg/DG-Design-Sequence-Diagram.png?raw=true "CreateCommand UML
```
You can consider terminating CreateCommand

![image](https://user-images.githubusercontent.com/60382274/97516099-5c274680-19cd-11eb-93e2-c6a6844b1052.png)

kaijiel24 replied:
@zhixiangteoh

zhixiangteoh replied:
Won't terminate CreateCommand here because this just shows the creation of CreateCommand, returning `createCommand` object whose sole purpose of life is to have its `execute()` method called, but in this sequence diagram that is essentially a snippet of the lifeline of `createCommand` I cannot also include its interactions with Main class because that's out of scope here; the point is that `createCommand` does exist for the whole duration shown here. 

Although, I should include some textual description of the gist of what I am conveying here, thanks!

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+`toString(boolean isQuestion, int offset)` method of a `FlashCard`
+object.
+
+![DG-Implementation-Features-PP-Sequence](./images-dg/PP-Sequence.png?raw=true)
```
Are you labelling correct? Perhaps it should be toString(isQuestion, offset) and formatResponse(isQuestion, offset)

![image](https://user-images.githubusercontent.com/60382274/97516241-aa3c4a00-19cd-11eb-8e3e-5718f4eec84a.png)

kaijiel24 replied:
@LiewWS

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+The user can also modify to tags of the decks by using tag or untag command, and uses search by tag to find
+a group of decks he/she is interested in.
+
+![DG-Implementation-Features-TagSequence](./images-dg/Tag.png?raw=true)
```
Are you missing a return arrow here?

![image](https://user-images.githubusercontent.com/60382274/97514907-d4d8d380-19ca-11eb-8d65-cd726d3459e0.png)

zhixiangteoh replied:
> A minor comment: For the sequence diagram below, would it be better if `Main` class is put on the left-most side? Because I misunderstood it first.
> ![image](https://user-images.githubusercontent.com/42442406/97528542-f779e500-19e8-11eb-8991-f3e0f5243711.png)

I believe the idea we were trying to convey is that the user does not interact directly with the `Main` class, but directly with the `Ui` class instead via the `Ui`'s `readUserInput()` method.

siewwencode commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+The following elaborates the execution flow of Game Mode, from after a `start` command has been parsed in Normal Mode:
+
+![DG-Implementation-Features-Game-Mode-Sequence](./images-dg/Game-Mode-Sequence.png?raw=true "Game Mode UML Sequence
```
some return arrows are missing

![image](https://user-images.githubusercontent.com/60382274/97516571-58e08a80-19ce-11eb-90b4-0d4385ccc1c1.png)

zhixiangteoh replied:
Thanks for the input! Return arrows are not mandatory, so this was intentional.

siewwencode reviewed (commented):
Overall the DG diagrams looks engaging in a sense that the diagrams are colored. The text explanation is clear and easy to understand.
However, you can consider having different markdowns for class and methods.

samuellleow commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+The user can also modify to tags of the decks by using tag or untag command, and uses search by tag to find
+a group of decks he/she is interested in.
+
+![DG-Implementation-Features-TagSequence](./images-dg/Tag.png?raw=true)
```
![image](https://user-images.githubusercontent.com/60358173/97518506-45cfb980-19d2-11eb-9fe2-745c2a39e540.png)

I believe that you can terminate the Ui class after printing the required statement

kaijiel24 replied:
@alwaysnacy

samuellleow reviewed (commented):
Overall great design and explanation of the DG, do consider and review the mistakes I pointed.

brandonywl commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+### Logic
+#### Overall Logic
+![DG-Overall Logic UML](./images-dg/Logic-DG.png?raw=true "Overall Logic Diagram")
```
![image](https://user-images.githubusercontent.com/28587601/97528176-09a75380-19e8-11eb-9be8-26f7de804cd8.png)

Unclear whether you are indicating they are all interactions or are some of them supposed to indicate inheritence?

Use triangle and arrows to differentiate between them.

kaijiel24 replied:
@alwaysnacy

brandonywl commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+#### Commands
+
+![DG-Design Commands UML](./images-dg/DG-Design-Commands.png?raw=true "Commands UML Class Diagram")
```
![image](https://user-images.githubusercontent.com/28587601/97528328-69056380-19e8-11eb-9946-94f2dd2186ee.png)

Triangles, not arrows, should be used to indicate inheritance of abstract classes

kaijiel24 replied:
@zhixianteoh

zhixiangteoh replied:
Changed, thanks!

t170815518 reviewed (commented):
A minor comment: For the sequence diagram below, would it be better if `Main` class is put on the left-most side? Because I misunderstood it first. 
![image](https://user-images.githubusercontent.com/42442406/97528542-f779e500-19e8-11eb-8991-f3e0f5243711.png)

brandonywl commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
```
![image](https://user-images.githubusercontent.com/28587601/97528532-f0eb6d80-19e8-11eb-8d71-d412e6a2a14f.png)

I believe that you should add a cross at the end of the life for the NormalCommand object. Furthermore, I believe from your logic class diagram below, you state that the parser is the one that creates the Command object. I take this to mean you instantiate a new Command object. In this case, you should point the arrow of execute to the constructor of the NormalCommand object.

This is found in your later diagrams.
![image](https://user-images.githubusercontent.com/28587601/97528712-550e3180-19e9-11eb-859e-9c30ee32b83d.png)

kaijiel24 replied:
@kaijiel24

brandonywl commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+   interacts with `GameStorage` on package-private basis; i.e., `GameEngine` and `GameStorage` have full mutual
+    access as if they were a single class. This is one of the main intentional design decisions.
+  
+![DG-Implementation-Features-Game-Mode-Architecture](./images-dg/Game-Mode-Design.png?raw=true "Game Mode
```
![image](https://user-images.githubusercontent.com/28587601/97529031-0745f900-19ea-11eb-8eaf-afd619f15123.png)

Is the GameStartCommand object still kept in reference? Will it still be used later? It is not indicated in the sequence diagram and an X can be added.

kaijiel24 replied:
@zhixiangteoh

zhixiangteoh replied:
It is actually not! Good catch! Changed, thanks!

brandonywl commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+The user can also modify to tags of the decks by using tag or untag command, and uses search by tag to find
+a group of decks he/she is interested in.
+
+![DG-Implementation-Features-TagSequence](./images-dg/Tag.png?raw=true)
```
![image](https://user-images.githubusercontent.com/28587601/97529236-6f94da80-19ea-11eb-8624-8e75a12a2ae2.png)

Random arrow that is not common notation.

kaijiel24 replied:
@alwaysnacy

brandonywl reviewed (commented):
Mostly notation errors.

t170815518 reviewed (commented):
**Parser**
I feel the class inheritance of parser could be added, just like `Commands` 
It may be better if more general explanations on `parseCommand` method,  e.g. what's your logic when writing `parseCommand` method, since it is the core part of parser. 
![image](https://user-images.githubusercontent.com/42442406/97530662-93a5eb00-19ed-11eb-83ea-af8697ad25aa.png)

t170815518 commented:
I am a bit confused what if `isPptxCreated` is `False` in this case? Will a new pptx be created ? 
![image](https://user-images.githubusercontent.com/42442406/97529915-f1393800-19eb-11eb-88df-d58ecbc0600d.png)

t170815518 commented:
**Overall Logic** 
The diagram seems not including all parsers and commands
Maybe the workflow arrow between `main`, `parser` and `main`, `command` should distinguish from the class-inheritance arrow? Because these two arrows are the same (shown below), it may be a bit confusing when first reading the diagram. 
![image](https://user-images.githubusercontent.com/42442406/97530596-7113d200-19ed-11eb-8d3e-4580e8f15a70.png)
![image](https://user-images.githubusercontent.com/42442406/97530927-12028d00-19ee-11eb-8698-f51380916cab.png)

t170815518 commented:
A general comment is maybe to add more explanations on exceptions

TomLBZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+### Logic
+#### Overall Logic
+![DG-Overall Logic UML](./images-dg/Logic-DG.png?raw=true "Overall Logic Diagram")
```
Should these arrows be triangular if you are trying to symbolize inheritance?
![image](https://user-images.githubusercontent.com/26886705/97535219-45e1b080-19f6-11eb-9ddc-59e6a8336b21.png)

TomLBZ replied:
Should this part be triangular arrow if you means inheritance?
![image](https://user-images.githubusercontent.com/26886705/97536080-960d4280-19f7-11eb-833b-a47962468a16.png)

TomLBZ reviewed (commented):
Some comments:)

JunxianAng commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+and search for relevant decks related to one or more relevant subjects in a robust way. Each created deck will
+tagged to their respective field.
+
+![DG-Implementation-Features-TagArchitecture](./images-dg/TagFeature-UML.png?raw=true)
```
![image](https://user-images.githubusercontent.com/50763491/97543877-cf4baf80-1a03-11eb-9959-ff7f2b37e86a.png)
minor issue but the overlapping arrows here make it seems a bit confusing. can be nicer

JunxianAng commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+#### Commands
+
+![DG-Design Commands UML](./images-dg/DG-Design-Commands.png?raw=true "Commands UML Class Diagram")
```
![image](https://user-images.githubusercontent.com/50763491/97546351-4c2c5880-1a07-11eb-8bfb-27b0394e8062.png)
Hi as all the arrow head are the same, the relationship between Nomral and Deck parser is confusing. Is it inheritance too?

JunxianAng commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,277 @@
+
+The following elaborates the execution flow of Game Mode, from after a `start` command has been parsed in Normal Mode:
+
+![DG-Implementation-Features-Game-Mode-Sequence](./images-dg/Game-Mode-Sequence.png?raw=true "Game Mode UML Sequence
```
Hi is objects like game:Game still kept in reference and used at the end? Can object deletion be used?

kaijiel24 replied:
@zhixiangteoh

teohdd committed:
update ug with proposed feature

zhixiangteoh committed:
Merge branch 'master' into master

zhixiangteoh committed:
Merge pull request #124 from zhixiangteoh/master

update ug with proposed feature

alwaysnacy committed:
add examples

wangwaynesg committed:
Add Class and Method conventions in DG

wangwaynesg committed:
Merge pull request #153 from wangwaynesg/master

Add Class and Method conventions in DG

wangwaynesg committed:
Add checking for corrupted Storage file

wangwaynesg committed:
Merge pull request #155 from wangwaynesg/master

Add checking for corrupted Storage file

LiewWS committed:
update .gitignore

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

teohdd committed:
update ug help commands, and command summary

teohdd committed:
add update command to command summary

teohdd committed:
update toc

zhixiangteoh committed:
Merge pull request #156 from zhixiangteoh/master

Update UG Command Summary, `help` documentation, TOC

teohdd committed:
fix wild 0 in deck's start command

wangwaynesg committed:
Fix Storage check corruption bugs

LiewWS committed:
Merge pull request #157 from zhixiangteoh/master

fix wild 0 in deck's start command

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

wangwaynesg committed:
Merge pull request #159 from wangwaynesg/master

Fix Storage check corruption bugs

LiewWS committed:
Fix bugs discovered in PED

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

LiewWS committed:
Refactor method for pretty printing to Ui

zhixiangteoh committed:
Merge pull request #160 from LiewWS/master

Fix bugs related to null question during semi-one-line add command

teohdd committed:
add zhixiangteoh tentative PPP

zhixiangteoh committed:
Merge pull request #161 from zhixiangteoh/master

add zhixiangteoh tentative PPP

teohdd committed:
update AboutUs links

teohdd committed:
update AboutUs PPP links

zhixiangteoh committed:
Merge pull request #162 from zhixiangteoh/master

Update AboutUs PPP links

teohdd committed:
add GameEngine, GameParser, GameStorage tests

zhixiangteoh committed:
Merge pull request #163 from zhixiangteoh/master

add GameEngine, GameParser, GameStorage tests

Commit:
Fixed some PE-D and Peer Review DG bugs

Commit:
Fixed index error messages

zhixiangteoh commented:
> **Parser**
> I feel the class inheritance of parser could be added, just like `Commands`
> It may be better if more general explanations on `parseCommand` method, e.g. what's your logic when writing `parseCommand` method, since it is the core part of parser.
> ![image](https://user-images.githubusercontent.com/42442406/97530662-93a5eb00-19ed-11eb-83ea-af8697ad25aa.png)

`parseCommand()` **is** one of our main methods, that is indeed extremely important in the logic of our program! Thanks for this suggestion, will include a section in DG!

zhixiangteoh commented:
> A general comment is maybe to add more explanations on exceptions

This is also an extremely good suggestion that we will make sure to include in our final DG :)

Commit:
Update Text-Ui-Test

kaijiel24 committed:
Merge pull request #164 from kaijiel24/master

Fixed some PE-D and Peer Review DG bugs

LiewWS committed:
update add flashcard command description

LiewWS committed:
update DG for pretty print

LiewWS committed:
update DG for Ui

teohdd committed:
update Commands, Game Mode sections, add Exceptions section to DG

zhixiangteoh committed:
Merge pull request #166 from zhixiangteoh/master

update Commands, Game Mode sections, add Exceptions section to DG

zhixiangteoh committed:
Merge branch 'master' into master

teohdd committed:
update PPP

zhixiangteoh committed:
Merge pull request #167 from zhixiangteoh/master

update PPP

LiewWS committed:
Update PPP

LiewWS committed:
Merge pull request #165 from LiewWS/master

Update UG and DG according to feedback.

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

LiewWS committed:
Update PPP

LiewWS committed:
Merge branch 'master' of https://github.com/LiewWS/tp into master

LiewWS committed:
Merge pull request #168 from LiewWS/master

Update PPP

teohdd committed:
update DG with class and method font styles, update PPP

zhixiangteoh committed:
Merge pull request #169 from zhixiangteoh/master

update DG with class and method font styles, update PPP

teohdd committed:
shorten PPP UG and DG extracts

zhixiangteoh committed:
Merge pull request #170 from zhixiangteoh/master

shorten PPP UG and DG extracts

teohdd committed:
fix bug where entering just punctuation as answer can lead to 100.0 score

teohdd committed:
add unit tests for getMatchPercentage

teohdd committed:
add functional and test code for extra restrictions on question and answer to contain alphanumeric

LiewWS committed:
Merge pull request #171 from zhixiangteoh/master

fix bug where entering just punctuation as answer leads to 100.0 score

LiewWS committed:
implement pretty print for game mode output

LiewWS committed:
Implement log to file

LiewWS committed:
update .gitignore

kaijiel24 committed:
Merge pull request #172 from LiewWS/master

Pretty Printing in Game Mode and Log to File

teohdd committed:
cut PPP UG and DG extracts images

zhixiangteoh committed:
Merge pull request #173 from zhixiangteoh/master

cut PPP UG and DG extracts images

Commit:
Add Original Color option for pptx

Commit:
Merged with upstream

Commit:
Fixed logger merge issue in NormalParser and Ui

Commit:
Add Default Color Scheme options to pptx

Commit:
Update UG for pptx Color options

Commit:
Merged from master

Commit:
Fixed length issue in help command

LiewWS committed:
Merge pull request #175 from kaijiel24/master

Added Color options for Print to PowerPoint command

LiewWS committed:
Fix index out of bounds error when -cs option is used

kaijiel24 committed:
Merge pull request #176 from LiewWS/master

Fix index out of bounds error when -cs option is used

Commit:
Fixed help printing issue

Commit:
Fixed help printing issue

wangwaynesg committed:
Add PPP for wangwaynesg

kaijiel24 committed:
Merge pull request #177 from kaijiel24/master

Fixed help printing issue

LiewWS committed:
fix broken links to commands in UG

LiewWS committed:
update FAQ section of UG

LiewWS committed:
Merge pull request #180 from LiewWS/master

User Guide updates

wangwaynesg committed:
Merge branch 'branch-wangwaynesgPPP' of https://github.com/wangwaynesg/tp

wangwaynesg committed:
Add wangwaynesg PPP

LiewWS committed:
update PPP

LiewWS committed:
update documentation

LiewWS committed:
Merge pull request #182 from LiewWS/master

Updates to UG, DG and PPP

LiewWS committed:
Merge pull request #181 from wangwaynesg/master

Add wangwaynesg PPP

alwaysnacy committed:
add DuplicateDeckException

alwaysnacy committed:
fix duplicate decks

alwaysnacy committed:
fix multiple /tags in CreateCommand

alwaysnacy committed:
update Tag and Untag Command

alwaysnacy committed:
fix checkstyles

alwaysnacy committed:
Merge pull request #183 from alwaysnacy/Trang-fix-2.0

Trang fix version 2.0

alwaysnacy committed:
add tag to DecksCommnad output without pretty 1-liner

alwaysnacy committed:
Merge pull request #184 from alwaysnacy/Trang-fix-2.0

add tag to DecksCommnad output without pretty 1-liner

alwaysnacy committed:
clean codes

alwaysnacy committed:
no message

alwaysnacy committed:
Merge pull request #185 from alwaysnacy/Trang-fix-2.0

clean codes

Commit:
Update kaijiel24 PPP

wangwaynesg committed:
Fix Storage bugs

wangwaynesg committed:
Merge pull request #186 from wangwaynesg/master

Fix Storage bugs

wangwaynesg committed:
Fix Storage bugs where decks with no tags are not loaded

wangwaynesg committed:
Fix Storage bugs where decks with no tags are not loaded and remove "size" from data files

wangwaynesg committed:
Fix List Command regarding incorrect "/ans" argument

wangwaynesg committed:
Fix minor Storage bug

wangwaynesg committed:
Merge pull request #206 from wangwaynesg/master

 Fix Storage bugs where decks with no tags are not loaded and remove "size" from data files

Commit:
Update pptx Color Options

Commit:
Resolve merge conflict

Commit:
Fix force y output and dash line to 80 char

kaijiel24 committed:
Merge pull request #207 from kaijiel24/master

Update pptx, fix delete force y output, increase dash line to 80 char and update UG and tests to reflect changes

LiewWS committed:
fix FlashCard test case

LiewWS committed:
Merge pull request #209 from LiewWS/master

fix FlashCard test case

Commit:
Update UG and DG

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

kaijiel24 committed:
Merge pull request #210 from kaijiel24/master

Update UG and DG

LiewWS committed:
update empty deck list message

LiewWS committed:
fix one-line add command

LiewWS committed:
update help command

LiewWS committed:
separate lines when updating flash card

LiewWS committed:
update UG

teohdd committed:
fixes #201

LiewWS committed:
update UG

teohdd committed:
fixes #204

teohdd committed:
fixes #212

teohdd committed:
fixes #190

teohdd committed:
fixes #203

teohdd committed:
fix checkstyle indentation

wangwaynesg committed:
Add Storage check for non-alphanumeric question and answer.
Check for duplicate deck names also.
Fix bug with empty tags data file loading.

wangwaynesg committed:
Fix minor checkstyle issue

alwaysnacy committed:
refine tag and untag

zhixiangteoh committed:
Merge pull request #213 from zhixiangteoh/master

Resolve Game Mode pretty print, Help commands pretty print, corresponding UG sections

zhixiangteoh committed:
Merge branch 'master' into master

LiewWS committed:
Merge pull request #214 from wangwaynesg/master

Add Storage check for non-alphanumeric question and answer.

LiewWS committed:
Update UG and DG

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

LiewWS committed:
update help in deck mode inUG

alwaysnacy committed:
furether refinement for tag and untag

LiewWS committed:
Merge pull request #215 from LiewWS/master

Fix issues discovered during meeting

alwaysnacy committed:
update UG

alwaysnacy committed:
update from remote

alwaysnacy committed:
Merge branch 'master' into Trang-v2.1

alwaysnacy committed:
Merge branch 'master' into Trang-v2.1

alwaysnacy committed:
fix checklists

alwaysnacy committed:
Merge branch 'Trang-v2.1' of https://github.com/alwaysnacy/tp into Trang-v2.1

alwaysnacy committed:
no message

alwaysnacy committed:
fix checkstyles

alwaysnacy committed:
remove MulitpleLabelInputException

alwaysnacy committed:
Merge pull request #216 from alwaysnacy/Trang-v2.1

Trang v2.1

teohdd committed:
add deque to glossary

teohdd committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

zhixiangteoh committed:
Merge pull request #217 from zhixiangteoh/master

add deque to glossary

teohdd committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

teohdd committed:
update DG manual testing instr. with game mode

alwaysnacy committed:
fix double spaces

alwaysnacy committed:
Merge pull request #221 from alwaysnacy/master

fix double spaces

alwaysnacy committed:
Merge pull request #220 from zhixiangteoh/master

add game mode manual testing instructions

alwaysnacy committed:
update UG

alwaysnacy committed:
Clean codes

Commit:
Update pptx DG, glossary, manual test, and JUnit Test

Commit:
Fixed DG merge conflict

wangwaynesg committed:
Update UG with information regarding editing the save data file

wangwaynesg committed:
Merge pull request #223 from wangwaynesg/master

Update UG with information regarding editing the save data file

wangwaynesg committed:
Fix minor UG issue where missing one link

wangwaynesg committed:
Merge pull request #224 from wangwaynesg/master

Fix minor UG issue where missing one link

alwaysnacy committed:
fix tagcommand

alwaysnacy committed:
add Parser diagram

alwaysnacy committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into Trang-addTests

LiewWS committed:
fix broken link to About US

LiewWS committed:
fix broken link to storage FAQ in UG

LiewWS committed:
update version number in case we forget later

LiewWS committed:
update ppp name to format required

LiewWS committed:
Rename LiewWS.md to liewws.md

wangwaynesg committed:
Fix image size

wangwaynesg committed:
Merge pull request #228 from wangwaynesg/master

Fix image size

LiewWS committed:
clarify requirements of adding question and answer in UG

LiewWS committed:
Block user from entering non-alphanumeric qn and ans in update

LiewWS committed:
update UserGuide to be specific about updating question and answers to non-alphanumeric characters

Commit:
Fixed typos in DG

wangwaynesg committed:
Update Storage format read and write

wangwaynesg committed:
Update UG to match changes in Storage format

wangwaynesg committed:
Merge pull request #229 from wangwaynesg/master

Update Storage format read and write

Commit:
Update picture size to fit within 100px

Commit:
Update PPP

kaijiel24 committed:
Merge pull request #222 from kaijiel24/master

Update pptx DG, glossary, manual testing, and JUnit Tests

LiewWS committed:
update profile picture

LiewWS committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

LiewWS committed:
update about us to include profile pic

Commit:
Updated PPP and DG

kaijiel24 committed:
Merge pull request #230 from kaijiel24/master

Updated PPP and DG

kaijiel24 committed:
Merge pull request #227 from LiewWS/master

Fix broken links and update version number

Commit:
Update PPP

kaijiel24 committed:
Merge pull request #231 from kaijiel24/master

Update PPP

Commit:
Update PPP

kaijiel24 committed:
Merge pull request #232 from kaijiel24/master

Update PPP

alwaysnacy committed:
update DG and UG

alwaysnacy committed:
Add files via upload

teohdd committed:
update PPP link

alwaysnacy committed:
update UG, DG and PPP

zhixiangteoh committed:
Merge pull request #233 from zhixiangteoh/master

update PPP link

alwaysnacy committed:
Merge branch 'alwaysnacy-Trang-addTests' into master

alwaysnacy committed:
no message

teohdd committed:
update toc dg ug

alwaysnacy committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp into master

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/images-dg/Tag-feature.png

zhixiangteoh committed:
Merge pull request #235 from zhixiangteoh/master

update toc dg ug

alwaysnacy committed:
no message

alwaysnacy committed:
Merge pull request #236 from alwaysnacy/master

Update DG and UG

Commit:
Fixed bugs for pptx in deck mode and updated UG and DG

Commit:
Fixed bugs for pptx in deck mode and updated UG and DG

wangwaynesg committed:
Update DG Storage sequence diagram to have transparent background

kaijiel24 committed:
Merge pull request #239 from kaijiel24/master

Fix pptx deck mode bug

wangwaynesg committed:
Merge pull request #240 from wangwaynesg/master

Update DG Storage sequence diagram to have transparent background

Commit:
Update DG diagram

kaijiel24 committed:
Merge pull request #241 from kaijiel24/master

Update DG diagram

LiewWS committed:
fix white bg images

LiewWS committed:
Merge pull request #242 from LiewWS/master

fix white bg images

teohdd committed:
fixes #237

teohdd committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-2/tp

teohdd committed:
update UG setup instructions

zhixiangteoh committed:
Merge pull request #243 from zhixiangteoh/master

fixes #237, fixes #244

alwaysnacy committed:
Add files via upload

alwaysnacy committed:
Add files via upload

alwaysnacy committed:
Delete alwaysnacy.png

alwaysnacy committed:
Add files via upload

alwaysnacy committed:
Delete alwaysnacy.png

alwaysnacy committed:
Add files via upload

alwaysnacy committed:
Delete alwaysnacy.png

alwaysnacy committed:
Add files via upload
